Allow user to handle phone account manually

jpudysz opened the pull request:
This small PR allows user to handle phone account flow manually.
Currently, the library is too obtrusive and it can be an issue for a well-designed app which doesn't want to display ugly Android alert.

What I'm proposing is another way for the developer to handle Android's phone account. Changes are backwards compatible so it can be released as a minor version.

How `react-native-callkeep` works now:
1. User can handle `microphone`, `camera` and `phone state` permissions manually. If user accepts it before `RNCallKeep.setup`, library won't display any prompts
2. After initial `setup` JS part of the lib will prompt the user to enable phone account and we can only inject alert's title and message
3. As a developer, I cannot change this behaviour

How library will work with this PR merged to master:
1. Same as in the master branch, we can request every permission beforehand
2. Developer can check if account permission is already enabled (eg. for visual representation) with new method `checkPhoneAccountEnabled`
3. Developer can register phone account manually (before `setup`) with new method `registerPhoneAccount` - with such option we can prompt the user and navigate to account settings manually, we can use our UI eg.:

<img src="https://user-images.githubusercontent.com/9088288/85038960-80edf080-b187-11ea-8b8b-0173df6b4a90.png" alt="example" width="300" />

4. At this point, we can `setup` library and enjoy CallKeep without any additional alerts or prompts ;)

manuquentin commented:
Thanks @jpudysz for the PR. I'll take a look.

Meanwhile, can you rebase it on master ?

manuquentin commented:
Moved in #242

jpudysz commented:
@manuquentin yup, looks good. It will improve UX a lot.
When it comes to the `backToForeground` it can be improved more.
I successfully launched my app last week. I created additional improvements to a headless mode like passing `callUUID` and consuming it after a cold boot. I used it to redirect to my custom WebRTC call UI so I'm not sure it's a common case for the community.

jdegger commented:
This PR is great, when is this due to be merged?

sboily commented:
@jdegger we need to have a rebase from @jpudysz and review to merge it.

jpudysz commented:
Hello everyone, I'm super busy and I can't contribute here, what's more, we are far behind each other. You can check my repo with a few additional improvements for Android (https://github.com/jpudysz/react-native-callkeep) - tested on production as I'm developing for my client's VoIP app.

I tried to create PR here, but I changed backToForeground logic and it differs from original repo. New functionality such as headless mode, locked screen are strictly connected with this function and I can't merge it. Ideally, we can take my ideas and merge it with the master (I don't mind and you can copy-paste my code under your PR). Ping me if I needed.

PS. I added changelog inside my repo, so you can check new functionalities.
@manuquentin if you want to discuss how to merge it or you need some explanation ping/PM me.
